File: iOS/AttendanceTracker/AttendanceTracker/Models/AttendanceRecord.swift
import Foundation
import SwiftData

/// Represents the status of attendance
enum AttendanceStatus: String, Codable {
    case present = "PRESENT"
    case absent = "ABSENT"
    case noClass = "NO_CLASS"
}

/// Represents an attendance record for a specific subject on a specific date
@Model
final class AttendanceRecord {
    @Attribute(.unique) var id: UUID
    var subjectId: UUID
    var date: Date
    var status: AttendanceStatus
    
    init(
        id: UUID = UUID(),
        subjectId: UUID,
        date: Date,
        status: AttendanceStatus
    ) {
        self.id = id
        self.subjectId = subjectId
        self.date = date
        self.status = status
    }
}

File: app/src/main/java/com/attendance/tracker/data/model/AttendanceRecord.kt
package com.attendance.tracker.data.model

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey
import java.time.LocalDate

/**
 * Represents an attendance record for a specific subject on a specific date
 */
@Entity(
    tableName = "attendance_records",
    foreignKeys = [
        ForeignKey(
            entity = Subject::class,
            parentColumns = ["id"],
            childColumns = ["subjectId"],
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [Index(value = ["subjectId", "date"], unique = true)]
)
data class AttendanceRecord(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val subjectId: Long,
    val date: LocalDate,
    val status: AttendanceStatus,
    val count: Int = 1
)

enum class AttendanceStatus {
    PRESENT,
    ABSENT,
    NO_CLASS
}
